Tasks/JavaToolInstallerV0/FileExtractor/JavaFilesExtractor.ts

```diff
@@ -148,9 +172,19 @@ export class JavaFilesExtractor {
         const jdkFile = path.normalize(repoRoot);
         const stats = taskLib.stats(jdkFile);
         if (stats.isFile()) {
-            await this.extractFiles(jdkFile, fileEnding)
+            await this.extractFiles(jdkFile, fileEnding);
             finalDirectoriesList = taskLib.find(this.destinationFolder).filter(x => taskLib.stats(x).isDirectory());
-            jdkDirectory = finalDirectoriesList.filter(dir => initialDirectoriesList.indexOf(dir) < 0)[0];
+            const freshExtractedDirectories = finalDirectoriesList.filter(dir => initialDirectoriesList.indexOf(dir) < 0);
```

Comment:
Can we add some more comments explaining what this does a little better please? I see that we are looking at the extracted directories looking for the bin directory, but then I'm a little confused by the if/else logic below

Reply:
Fair enough. Usage of De Morgans's law made this expression a little complicated.
I tried to refactor this part to make it easier to understand.